src/Tribe/Views/V2/View.php

```diff
@@ -2145,4 +2154,92 @@ protected function get_container_data() {
 
 		return $data;
 	}
+
+	protected function should_show_recent_past_events_view() {
+		$show = $this->context->get( 'show_recent_past', true );
+
+		/**
+		 * Filters Whether the Recent Past Events Should Show for all Views.
+		 *
+		 * @since TBD
+		 *
+		 * @param aboolean $show      If we should display Recent Past Events.
+		 * @param string   $view_slug The current view slug.
+		 * @param View     $instance  The current View object.
+		 */
+		$show = apply_filters( 'tribe_events_views_v2_show_recent_past_events_view', $show, $this->get_slug(), $this );
+
+		/**
+		 *  Filters Whether the Recent Past Events Should Show for a specific View.
+		 *
+		 * @since TBD
+		 *
+		 * @param boolean $show     If we should display Recent Past Events.
+		 * @param View    $instance The current View object.
+		 */
+		$show = apply_filters( "tribe_events_views_v2_{$this->get_slug()}_show_recent_past_events_view", $show, $this );
+
+		return $show;
+	}
+
+
+	/**
+	 * @param array $events
+	 * @param array $template_vars
+	 */
+	protected function setup_additional_views( array $events = [], array $template_vars = [] ) {
+		// todo: flatten the events if required (month, week)
+		// todo add threshold and filter
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 685e6ce1f24cf120bf3bc60d1ce7bc43</sub>